core/sis-feature/src/test/java/org/apache/sis/feature/builder/FeatureTypeBuilderTest.java

```diff
@@ -344,4 +347,95 @@ private static void assertPropertyEquals(final String name, final Class<?> value
         }
         assertEquals("valueClass", valueClass, ((DefaultAttributeType<?>) property).getValueClass());
     }
+
+    @Ignore("Obviously a bug.")
```

Comment:
Actually the `NullPointerException` come from this test invoking `setSuperTypes` with an array of null value. The fix is not in `toStringInternal`, but rather in `setSuperTypes` which should check for null argument values.